apm-agent-core/src/main/java/co/elastic/apm/agent/impl/circuitbreaker/CircuitBreaker.java

```diff
@@ -108,6 +115,41 @@ private void pollStressMonitors() {
         }
     }
 
+    private void checkIfUnderStress() {
```

Comment:
It seems a bit strange that a method named `check*` modifies state. Could we just return a boolean here and handle based on that in `pollStressMonitors`?

```
if (isUnderStress()) {
    isCurrentlyUnderStress = true;
    tracer.stressDetected();
}
```

Reply:
Yes we can